src/backend/distributed/commands/utility_hook.c

```diff
@@ -632,6 +637,32 @@ citus_ProcessUtilityInternal(PlannedStmt *pstmt,
 		{
 			ddlJobs = ops->preprocess(parsetree, queryString, context);
 		}
+
+		if (ddlJobs != NIL)
+		{
+			List *nonCoordinatorNodeJobs = NIL;
+
+			DDLJob *ddlJob = NULL;
+			foreach_ptr(ddlJob, ddlJobs)
+			{
+				DDLJob *coordinatorJob = NULL;
+				DDLJob *nonCoordinatorNodeJob = NULL;
+				SplitCoordinatorJob(ddlJob, &coordinatorJob, &nonCoordinatorNodeJob);
+
+				if (coordinatorJob)
+				{
+					ExecuteDistributedDDLJob(coordinatorJob);
```

Comment:
> if a command fails, you get the error on the table instead of on the shards -- which has been very confusing.

That's indeed a good goal, but I don't think that's a big issue as long as we send the actual metadata updating DDL command first to the coordinator (instead of a shard command). You'd still get the "while running on coordinator.some.host.com:5432", but that seems rather harmless.

> Similarly, if there are permission problems or any other problems, Citus does not use any resources (e.g., connections) before doing things locally.

I don't think this is an issue we really need to worry about. If an attacker is trying to exhaust resources, then this does not seem like a very smart way of doing so (e.g. run_command_on_workers + generate_series + cross join seems a much easier way to bring stuff down). And if it's by accident I don't think opening a connection is a huge resource hog, and probably a connection is already cached anyway.

Reply:
> That's indeed a good goal, but I don't think that's a big issue as long as we send the actual metadata updating DDL command first to the coordinator (instead of a shard command). You'd still get the "while running on coordinator.some.host.com:5432", but that seems rather harmless.

That caused quite a bit of confusion for the users in the past, so I would not underestimate impact of that UX change